src/Microsoft.TestPlatform.Utilities/CodeCoverageDataAttachmentsHandler.cs

```diff
@@ -0,0 +1,106 @@
+
+        private static readonly string[] SupportedFileExtensions = { ".dll", ".exe" };
+
+        public ICollection<AttachmentSet> HandleDataCollectionAttachmentSets(ICollection<AttachmentSet> dataCollectionAttachments)
+        {
+            if (dataCollectionAttachments == null)
+                return new Collection<AttachmentSet>();
+
+            var coverageAttachments = dataCollectionAttachments
+                .Where(dataCollectionAttachment => CodeCoverageDataCollectorUri.Equals(dataCollectionAttachment.Uri)).ToArray();
+
+            if (coverageAttachments.Any())
+            {
+                var codeCoverageFiles = coverageAttachments.Select(coverageAttachment => coverageAttachment.Attachments[0].Uri.LocalPath).ToArray();
+                var outputFile = MergeCodeCoverageFiles(codeCoverageFiles);
+                var attachmentSet = new AttachmentSet(CodeCoverageDataCollectorUri, CoverageFriendlyName);
+
+                if (!string.IsNullOrEmpty(outputFile))
+                {
+                    attachmentSet.Attachments.Add(new UriDataAttachment(new Uri(outputFile), CoverageFriendlyName));
+                    return new Collection<AttachmentSet> { attachmentSet };
+                }
+                // In case merging fails(esp in dotnet core we cannot merge), so return filtered list of Code Coverage Attachments
+                return coverageAttachments;
+            }
+
+            return new Collection<AttachmentSet>();
+        }
+
+        private string MergeCodeCoverageFiles(IList<string> files)
+        {
+            string fileName = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString() + CoverageFileExtension);
+            string outputfileName = files[0];
+
+            File.Create(fileName).Dispose();
+
+            foreach (var extension in SupportedFileExtensions)
+            {
+                var assemblyPath = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), CodeCoverageAnalysisAssemblyName + extension);
+
+                try
+                {
+                    Assembly assembly = null;
+#if NET46
+                    assembly = Assembly.LoadFrom(assemblyPath);
+#else
+                    assembly = AssemblyLoadContext.Default.LoadFromAssemblyPath(assemblyPath);
+#endif
+                    var type = assembly.GetType(CodeCoverageAnalysisAssemblyName + "." + CoverageInfoTypeName);
+
+                    var methodInfo = type?.GetMethod(MergeMethodName);
+
+                    if (methodInfo != null)
+                    {
+                        for (int i = 1; i < files.Count; i++)
+                        {
+                            methodInfo.Invoke(null, new object[] { files[i], outputfileName, fileName, true });
+                            File.Copy(fileName, outputfileName, true);
+
+                            File.Delete(files[i]);
+                        }
+
+                        File.Delete(fileName);
```

Comment:
Suggestion : File operations can be done through FileHelper (IFileHelper) interface, this will help in writing test cases.